### PR TITLE
Add deduplication

### PR DESCRIPTION
**Pull request by hamdielhamdi-cognite**
## Checklist

- [x] Changelog entry added in `CHANGELOG` or not relevant for this change
- [x] Version updated in `pyproject.toml` or not relevant for this change

Fix/Add alert tests.
Add alert deduplication endpoint.

### CODE REVIEW

File: cognite/experimental/_api/alerts.py [hunk 1]
```diff
@@ -193,6 +200,18 @@ def create(
             items=alerts, resource_path=self._RESOURCE_PATH, list_cls=AlertList, resource_cls=Alert
         )
 
+    def create_deduplicate(
```
Comment (polomani) [inline cognite/experimental/_api/alerts.py:203]:
```suggestion
    def create_deduplicated(
```
possibly?

File: tests/tests_integration/test_alerts/test_alerts.py [hunk 2]
```diff
@@ -87,16 +141,18 @@ def create(channel_id: int, subscriber_id: int) -> AlertSubscription:
 class TestAlertChannelsIntegration:
     def test_create_1(self, cognite_client, base_channel):
         res = cognite_client.alerts.channels.create(base_channel())
-        assert "test_" in res.external_id
+        "test_" in res.external_id
```
Comment (polomani) [inline tests/tests_integration/test_alerts/test_alerts.py:144]:
```suggestion
        assert "test_" in res.external_id
```

File: tests/tests_integration/test_alerts/test_alerts.py [hunk 3]
```diff
@@ -43,6 +61,28 @@ def create(channel_external_id: int) -> Alert:
     return create
 
 
+@fixture(scope="class")
+def base_alert_with_deduplication_rules() -> Callable[..., Alert]:
```
[the remark this answers is not in the document]
Comment (polomani) [reply tests/tests_integration/test_alerts/test_alerts.py:65]: deduplication rule is something on the channel, so this name is more explanatory

File: CHANGELOG.md [hunk 4]
```diff
@@ -12,6 +12,12 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [0.108.0]
+
+### Fixed
```
Comment (polomani) [inline CHANGELOG.md:17]:
```suggestion
### Added
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -12,6 +12,12 @@ Changes are grouped as follows
...
+- Add deduplication endpoint to the alert module.
```
Comment (polomani) [inline CHANGELOG.md:18]:
```suggestion
- Add deduplication endpoint to the alert module.

### Fixed
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -12,6 +12,12 @@ Changes are grouped as follows
...
+- Fix existing tests and Add new tests.
```
Comment (polomani) [inline CHANGELOG.md:19]:
```suggestion
- Fix existing alerts api tests and add new tests. 
```

File: cognite/experimental/_api/alerts.py [hunk 7]
```diff
@@ -193,6 +200,18 @@ def create(
...
+        self,
+        alerts: Union[Alert, List[Alert]],
+    ) -> Union[Alert, AlertList]:
+        assert_type(alerts, "alerts", [Alert, list])
+        return self._create_multiple(
+            items=alerts,
+            resource_path=self._RESOURCE_PATH_DEDUPLICATE,
+            list_cls=DeduplicateAlertList,
+            resource_cls=DeduplicateAlert,
```
Comment (polomani) [inline cognite/experimental/_api/alerts.py:212]: pretty sure list_cls and resource_cls are response types? then it should be the same as for "create" endpoint.

but the input should be DeduplicateAlert instead:
```
         alerts: Union[DeduplicateAlert, List[DeduplicateAlert]],
         
         ....
assert_type(alerts, "alerts", [DeduplicateAlert, list])
```

File: tests/tests_integration/test_alerts/test_alerts.py (hunk 2 above)
Comment (hamdielhamdi-cognite) [reply tests/tests_integration/test_alerts/test_alerts.py:144]: done

File: CHANGELOG.md (hunk 6 above)
Comment (hamdielhamdi-cognite) [reply CHANGELOG.md:19]: done

**Review by polomani — APPROVED** [review]

**Pull request merged**